### DISCUSSION TITLE
Problems with InitialViewChange & marker

### DISCUSSION THREAD

**Discussion by Ahmderk** [uber/react-map-gl#1868] [Q&A]
Hey guys, i need some help.

I set up a map with a marker on it, but the marker it is not fixed, i mean when i zoom with scroll button in my mouse the marker will move around. down below i will give you my code

`import React from 'react'
import Map, {Marker, Popup} from 'react-map-gl';
import getCenter from 'geolib/es/getCenter'

function Maps({searchResults}) {

    const coordinates = searchResults.map((result) => ({
        longitude: result.long,
        latitude: result.lat
    }))

    const center = getCenter(coordinates);

    return (
        <Map
            mapStyle='mapbox://styles/ahmderk/
            cl3d5xb6h002914rndx4bc3z3'
            mapboxAccessToken={process.env.mapbox_key}
            initialViewState={{
                longitude:center.latitude,
                latitude:center.longitude,
                zoom: 11
            }}
        >
            {searchResults.map((result) => (
                <div key={result.long}>
                    <Marker
                        longitude={result.long}
                        latitude={result.lat}
                        offsetLeft={-20}
                        offsetTop={-10}
                    > 
                    <p className='cursor-pointer text-2xl animation-bounce'>🏘️</p>
                    </Marker>
                </div>

            ))}
        </Map>
    )
};

export default Maps

`

is there any wrong in there? thanks

**Comment by Pessimistress** [comment]
`offsetLeft` and `offsetTop` are not props supported by the current release. See https://visgl.github.io/react-map-gl/docs/api-reference/marker
